site/cds_rdm/inspire_harvester/writer.py

```diff
@@ -30,81 +32,126 @@ def _write_entry(self, entry, *args, **kwargs):
 
     def write_many(self, stream_entries, *args, **kwargs):
         """Creates or updates the record in CDS."""
         entries = [entry.entry for entry in stream_entries]
-
+        current_app.logger.info(
+            f"Writer entry point. Start processing a batch of stream entries: {entries}."
+        )
         for entry in entries:
             self._write_entry(entry, *args, *kwargs)
+        current_app.logger.info(f"Writing entries finished.")
         return stream_entries
 
     def _get_existing_records(self, inspire_id):
         """Find records that have already been harvested from INSPIRE."""
+        current_app.logger.info(
+            f"Start retrieving existing records by INSPIRE ID ({inspire_id})."
+        )
         # for now checking only by inspire id
         filters = [
             dsl.Q("term", **{"metadata.identifiers.scheme": "inspire"}),
             dsl.Q("term", **{"metadata.identifiers.identifier": inspire_id}),
         ]
         combined_filter = dsl.Q("bool", filter=filters)
+        current_app.logger.debug(
+            f"Filter for searching for existing records: {combined_filter}."
+        )
         return current_rdm_records_service.search(
             system_identity, extra_filter=combined_filter
         )
 
     def update_record(self, entry, record_pid):
         """Update existing record."""
+        current_app.logger.info(
+            f"Start updating an existing record on CDS. Recid: {record_pid}."
+        )
         record = current_rdm_records_service.read(system_identity, record_pid)
         record_dict = record.to_dict()
-
+        current_app.logger.debug(f"Existing record details: {record_dict}.")
         existing_files = record_dict["files"]["entries"]
+        current_app.logger.debug(f"Existing record's files details: {existing_files}.")
         new_files = entry["files"].get("entries", {})
 
         if not new_files:
-            # TODO log the absence of files
+            current_app.logger.error(
+                f"INSPIRE record #{entry['id']} has no files. Metadata-only records are not supported. Aborting record writer."
+            )
             return
-            # raise WriterError(f"INSPIRE record #{entry['id']} has no files. Aborting.")
 
         # Normalize the checksum format in existing for comparison
         existing_checksums = [
             value["checksum"] for key, value in existing_files.items()
         ]
+
+        current_app.logger.debug(f"Existing files' checksums: {existing_checksums}.")
         new_checksums = [value["checksum"] for key, value in new_files.items()]
+        current_app.logger.debug(f"New files' checksums: {new_checksums}.")
 
         should_create_new_version = existing_checksums != new_checksums
 
         if should_create_new_version:
+            current_app.logger.info(
+                "Differences between existing and new files checksums were found. Creating a new version for the record."
+            )
```

Comment:
I would put this as one message inside the _create_new_version method and report on the created version's PID